Sources/ProjectSpec/Scheme.swift

```diff
@@ -864,3 +876,34 @@ extension XCScheme.EnvironmentVariable: JSONEncodable {
         return dict
     }
 }
+
+extension XCScheme.LaunchAction.GPUFrameCaptureMode: JSONEncodable {
+    public func toJSONValue() -> Any {
+        switch self {
+        case .autoEnabled:
+            return "autoEnabled"
+        case .metal:
+            return "metal"
+        case .openGL:
+            return "openGL"
+        case .disabled:
+            return "disabled"
+        }
+    }
+
+    static func fromJSONValue(_ string: String) -> XCScheme.LaunchAction.GPUFrameCaptureMode {
+        switch string {
+        case "autoEnabled":
+            return .autoEnabled
+        case "metal":
+            return .metal
+        case "openGL":
+            return .openGL
+        case "disabled":
+            return .disabled
+        default:
+            assert(false, "Invalid enableGPUFrameCaptureMode value. Valid values are: autoEnabled, metal, openGL, disabled")
```

Comment:
i wonder if this should fail in some way that the user sees? otherwise they may not realize they have an invalid value, since this won't fail in release, what do other places like this do?

Reply:
looks like fatalError is preferred: https://github.com/search?q=repo%3Ayonaskolb%2FXcodeGen%20fatalError&type=code

Reply:
updated: https://github.com/yonaskolb/XcodeGen/pull/1251/commits/d5aa44faba4052e75f67330cdfed0078c871dd56